sklearn/tests/test_impute.py

```diff
@@ -1119,3 +1119,37 @@ def test_inconsistent_dtype_X_missing_values(imputer_constructor,
 
     with pytest.raises(ValueError, match=err_msg):
         imputer.fit_transform(X)
+
+
+@pytest.mark.parametrize("array_constr",
+                         [np.array, sparse.csr_matrix, sparse.csc_matrix],
+                         ids=["dense", "sparse_csr", "sparse_csc"])
+def test_missing_indicator_drop_full_missing(array_constr):
+    # Check that missing indicator with features="not-constant" drops columns
+    # with no missing values as well as columns full of missing values.
+    X = array_constr([[0, np.nan, 0],
+                      [0, np.nan, np.nan]])
+
+    expected_Xt = array_constr([[False],
+                                [True]])
+
+    mi = MissingIndicator(features="not-constant")
+    Xt = mi.fit_transform(X)
+
+    assert_allclose_dense_sparse(Xt, expected_Xt)
+
+
+def test_missing_indicator_sparse_no_explicit_zeros():
+    # Check that non missing values don't become explicit zeeros in the mask
+    # generated by missing indicator when X is sparse.
+    X = sparse.csr_matrix([[0, 1, 2],
+                           [1, 2, 0],
+                           [2, 0, 1]])
+    expected_nnz = 3  # 3 missing values
+
+    mi = MissingIndicator(features='all', missing_values=1)
+    Xt = mi.fit_transform(X)
+
+    nnz = Xt.getnnz()
+
```

Comment:
You could just add an assertion elsewhere that `Xt.nnz == Xt.sum()`. You shouldn't need a new test, nor a specified expected value.